core/store/src/trie/trie_recording.rs

```diff
@@ -133,43 +292,34 @@ mod trie_recording_tests {
     fn test_trie_recording_consistency_with_flat_storage(
```

Comment:
It looks like `test_trie_recording_consistency` in many places...  can we unite them and have tests with 4 bool args? :)
And well, I would just have a for-loop for a bitmask of parameters instead of 15 separate tests. Or make it a macros if we don't want to make one test very heavy.

Reply:
So the reason I don't want a loop that goes through the booleans is that when something fails, it just says the whole test fails and you'd have to dig through the logs to figure out which exact case failed. A macro solves this problem but I don't know how to write it 🤔